src/pages/academy/grading/subcomponents/GradingWorkspace.tsx

```diff
@@ -304,8 +304,16 @@ const GradingWorkspace: React.FC<GradingWorkspaceProps> = props => {
             initialXp={grading![questionId].grade.xp}
             xpAdjustment={grading![questionId].grade.xpAdjustment}
             maxXp={grading![questionId].question.maxXp}
-            studentName={grading![questionId].student.name}
-            studentUsername={grading![questionId].student.username}
+            studentName={
+              grading![questionId].student.name
+                ? grading![questionId].student.name
+                : grading![questionId].team?.map(member => member.name).join(', ')
+            }
+            studentUsername={
+              grading![questionId].student.username
+                ? grading![questionId].student.username
+                : grading![questionId].team?.map(member => member.username).join(', ')
+            }
```

Comment:
As mentioned above, ideally these two props should be an array. But if you want this, then you could just use short circuit i.e. `name || team!.map(...)`

Reply:
Changed!